flang/lib/Lower/OpenMP.cpp

```diff
@@ -716,6 +716,82 @@ genOMP(Fortran::lower::AbstractConverter &converter,
   createBodyOfOp<omp::CriticalOp>(criticalOp, converter, currentLocation, eval);
 }
 
+ void Fortran::lower::genOpenMPSectionsBlock(Fortran::lower::AbstractConverter &converter, Fortran::lower::pft::Evaluation &eval){
+     auto &firOpBuilder = converter.getFirOpBuilder(); 
+     auto currentLocation = converter.getCurrentLocation();
+     mlir::omp::SectionOp sectionOp = firOpBuilder.create<mlir::omp::SectionOp>(currentLocation);
+     createBodyOfOp<omp::SectionOp>(sectionOp, converter, currentLocation, eval);
+ }
+
+ // TODO: Add support for reduction
+ static void
+ genOMP(Fortran::lower::AbstractConverter &converter,
+        Fortran::lower::pft::Evaluation &eval,
+        const Fortran::parser::OpenMPSectionsConstruct &sectionsConstruct) {
+     auto &firOpBuilder = converter.getFirOpBuilder();
+     auto currentLocation = converter.getCurrentLocation();
+     Fortran::lower::StatementContext stmtCtx;
+     SmallVector<Value, 4> privateClauseOperands, firstPrivateClauseOperands,
+         lastPrivateClauseOperands, reductionVars, allocateOperands,
+         allocatorOperands;
+
+     const auto &sectionsClauseList = std::get<Fortran::parser::OmpClauseList>(
+             std::get<Fortran::parser::OmpBeginSectionsDirective>(sectionsConstruct.t).t);
+
+     for(const auto &clause : sectionsClauseList.v){ 
+         if (const auto &firstPrivateClause = std::get_if<Fortran::parser::OmpClause::Firstprivate>(&clause.u)){
+             const Fortran::parser::OmpObjectList &ompObjectList = firstPrivateClause->v;
+             genObjectList(ompObjectList, converter, firstPrivateClauseOperands);
+         }
+
+         if (const auto &privateClause = std::get_if<Fortran::parser::OmpClause::Private>(&clause.u)){
+             const Fortran::parser::OmpObjectList &ompObjectList = privateClause->v;
+             genObjectList(ompObjectList, converter, privateClauseOperands);
+         }
+
+         if (const auto &lastPrivateClause = std::get_if<Fortran::parser::OmpClause::Lastprivate>(&clause.u)){
+             const Fortran::parser::OmpObjectList &ompObjectList = lastPrivateClause->v;
+             genObjectList(ompObjectList, converter, lastPrivateClauseOperands);
+         }
```

Comment:
At the moment there is no privatisation handling in the OpenMP dialect. Can you look at how privatisation (lastprivate is not implemented) is implemented in parallel and do and do the same thing here?

Reply:
I compared the implementation of loop construct (`OpenMPLoopConstruct`) in the same file. For lastprivate, we are generating an object list of arguments (through `genObjectList`). For private and firstprivate, we are currently sending in empty lists to `fieOpBuilder.create(...)`